### DISCUSSION TITLE
Should request.url in prefixed plugin's handlers have prefix or not?

### DISCUSSION THREAD

**Discussion by lanyusan** [fastify/fastify#2913] [General]
I recently realized that even if a plugin has a prefix, say 'blog', and a request like `/blog/post/abc` will be passed to a router registered with the plugin, the router's hanlder's `request.url` still is the full url, `/blog/post/abc`. 

As the concept of prefixed plugin is to provide a sort of isolated transparent environment, shouldn't it be better that the `url` property have no prefix, just `/post/abc` ?

**Comment by jsumners** [comment]
`request.url` is the original `IncomingMessage` property.

**Reply by lanyusan** [reply]
Fastify's request has a property `raw`, which could get all the raw data. 

Prefixed plugin's routing rules don't need the prefix, which is some how inconsistent.
